src/errors.py

```diff
@@ -0,0 +1,72 @@
+# Copyright 2023 Canonical Ltd.
```

Comment:
style: Flask named this file as exceptions.py. What's the standard? Related https://github.com/canonical/jenkins-k8s-operator/pull/9#discussion_r1235250400

Reply:
`exceptions.py` is probably a better name.

Reply:
Minor comments